libosinfo.rb

```diff
@@ -0,0 +1,91 @@
+class Libosinfo < Formula
+  desc "The Operating System information database"
+  homepage "https://libosinfo.org/"
+  url "https://fedorahosted.org/releases/l/i/libosinfo/libosinfo-0.3.0.tar.gz"
+  sha256 "538a3468792e919edf5364fe102d751353ae600a92ad3a24f024424a182cefbc"
+
+  depends_on "intltool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "wget" => :build
+
+  depends_on "check"
+  depends_on "glib"
+  depends_on "libsoup"
+  depends_on "libxml2"
+  depends_on "libxslt"
+
+  depends_on "gobject-introspection" => :optional
+  depends_on "vala" => :optional
+
+  # work around "ld: unknown option: --no-undefined"
+  # upstream bug: https://bugzilla.redhat.com/show_bug.cgi?id=1305016
+  patch :DATA
+
+  def install
+    args = ["--prefix=#{prefix}",
+            "--localstatedir=#{var}",
+            "--mandir=#{man}",
+            "--sysconfdir=#{etc}",
+            "--disable-silent-rules",
+            "--disable-udev",
+            "--enable-tests"]
+
+    args << "--enable-introspection" if build.with? "gobject-introspection"
+    args << "--enable-vala" if build.with? "vala"
+
+    system "./configure", *args
+
+    # Compilation of docs doesn't get done if we jump straight to "make install"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    Pathname("test.py").write <<-EOS.undent
```

Comment:
addressed in https://github.com/Homebrew/homebrew/commit/5300bbc57216ad0fdb0203f1ce326975441325c7